roxie/ccd/ccdserver.cpp

```diff
@@ -1316,6 +1316,38 @@ class CRoxieServerActivity : implements CInterfaceOf<IRoxieServerActivity>, impl
             merged.mergeStatistic(StEnumActivityCharacteristics, (unsigned) cachedSourceCharacteristics.load());
     }
 
+    mutable unsigned lastLogTime = 0;
+    mutable unsigned minLogInterval = 10000;
+    mutable StringBuffer lastLogText;
+    mutable const char *lastLogFormat = nullptr;
+    mutable unsigned logSuppressed = 0;
+    mutable CriticalSection logCrit;
+
+    virtual void CTXLOGva(const LogMsgCategory & cat, const LogMsgJobInfo & job, LogMsgCode code, const char *format, va_list args) const override  __attribute__((format(printf,5,0))) 
+    {
+        StringBuffer text, prefix;
+        text.valist_appendf(format, args);
+        unsigned now = msTick();
+        CriticalBlock b(logCrit);
```

Comment:
Do we need to hold the crit during the actual printing/logging ?  Does it matter ??